src/commons_codec/transform/dynamodb.py

```diff
@@ -37,14 +39,45 @@ def _deserialize_ss(self, value):
     def _deserialize_bs(self, value):
         return list(super()._deserialize_bs(value))
 
+    def _deserialize_l(self, value):
+        """
+        CrateDB can't store varied lists in an OBJECT(DYNAMIC) column, so set the
+        stage to break them apart in order to store them in an OBJECT(IGNORED) column.
+
+        https://github.com/crate/commons-codec/issues/28
+        """
+
+        # Deserialize list as-is.
+        result = TaggableList([self.deserialize(v) for v in value])
+        result.set_tag("varied", False)
+
+        # If it's not an empty list, check if inner types are varying.
+        # If so, tag the result list accordingly.
+        # It doesn't work on the result list itself, but on the DynamoDB
+        # data structure instead, comparing the single/dual-letter type
+        # identifiers.
+        if value:
+            dynamodb_type_first = list(value[0].keys())[0]
+            for v in value:
+                dynamodb_type_current = list(v.keys())[0]
+                if dynamodb_type_current != dynamodb_type_first:
+                    result.set_tag("varied", True)
+                    break
+        return result
+
 
 class DynamoTranslatorBase:
     """
-    Translate DynamoDB CDC events into different representations.
+    Translate DynamoDB records into a different representation.
     """
 
-    # Define name of the column where CDC's record data will get materialized into.
-    DATA_COLUMN = "data"
+    # Define name of the column where typed DynamoDB fields will get materialized into.
+    # This column uses the `OBJECT(DYNAMIC)` data type.
+    TYPED_COLUMN = "data"
+
+    # Define name of the column where untyped DynamoDB fields will get materialized into.
+    # This column uses the `OBJECT(IGNORED)` data type.
+    UNTYPED_COLUMN = "aux"
```

Comment:
Running out of "naming things", please advise if you can come up with better labels.